dotCMS/src/main/java/com/dotcms/storage/S3Storage.java

```diff
@@ -0,0 +1,174 @@
+    }
+
+    private Host resolveHost () {
+
+        Host host = null;
+
+        final HttpServletRequest request = HttpServletRequestThreadLocal.INSTANCE.getRequest();
+        if (null != request) {
+            host = Try.of(() -> WebAPILocator.getHostWebAPI().getCurrentHostNoThrow(request)).getOrNull();
+        }
+
+        if (null == host) {
+            host = Try.of(() -> APILocator.getHostAPI().findDefaultHost(APILocator.systemUser(), false)).getOrNull();
+        }
+
+        return null == host ? APILocator.systemHost() : host;
+    }
+
+    @Override
+    public boolean existsBucket(final String bucketName) {
+
+        return this.getStorage().existsBucket(bucketName);
+    }
+
+    @Override
+    public boolean existsObject(final String bucket, final String objectPath) {
+
+        return false;
+    }
+
+    @Override
+    public boolean createBucket(final String bucketName) {
+
+        return Try.of(()-> { this.getStorage().createBucket(bucketName); return true; }).getOrElse(false);
+    }
+
+    @Override
+    public boolean createBucket(final String bucketName, final Map<String, Object> extraOptions) {
+        return Try.of(()-> {
+                            if (extraOptions.containsKey("region")) {
+                                this.getStorage().createBucket(bucketName, (String)extraOptions.get("region"));
+                            } else {
+                                this.getStorage().createBucket(bucketName);
+                            }
+                            return true;
+        }).getOrElse(false);
+    }
+
+    @Override
+    public boolean deleteBucket(final String bucketName) {
+        return Try.of(()-> {
+            this.getStorage().deleteBucket(bucketName);
+            return true;
+        }).getOrElse(false);
+    }
+
+    @Override
+    public boolean deleteObject(final String bucket, final String path) {
+        return Try.of(()-> {
+            this.getStorage().deleteFile(bucket, path);
+            return true;
+        }).getOrElse(false);
+    }
+
+    @Override
+    public List<Object> listBuckets() {
+        final List buckets = Try.of(()->this.getStorage().listBuckets()).getOrElse(()-> Collections.emptyList());
+        return buckets;
+    }
+
+    @Override
+    public Object pushFile(String bucketName, String path, File file, Map<String, Object> extraMeta) {
+        return null;
+    }
+
+    @Override
+    public Object pushObject(String bucketName, String path, ObjectWriterDelegate writerDelegate, Object object, Map<String, Object> extraMeta) {
+        return null;
+    }
+
+    @Override
+    public Future<Object> pushFileAsync(String bucketName, String path, File file, Map<String, Object> extraMeta) {
+        return null;
+    }
+
+    @Override
+    public Future<Object> pushObjectAsync(String bucketName, String path, ObjectWriterDelegate writerDelegate, Object object, Map<String, Object> extraMeta) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'writerDelegate' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5547873)